plugins/Installation/FormDatabaseSetup.php

```diff
@@ -98,23 +104,39 @@ function init()
             'type'          => $defaultDatabaseType,
             'tables_prefix' => 'matomo_',
             'schema'        => 'Mysql',
+            'port'          => '3306',
         );
 
-        $defaultsEnvironment = array('host', 'adapter', 'tables_prefix', 'username', 'schema', 'password', 'dbname');
+        $defaultsEnvironment = array('host', 'adapter', 'tables_prefix', 'username', 'schema', 'password', 'dbname', 'port');
         foreach ($defaultsEnvironment as $name) {
-            $envName = 'DATABASE_' . strtoupper($name); // fyi getenv is case insensitive
-            $envNameMatomo = 'MATOMO_' . $envName;
-            if (getenv($envNameMatomo)) {
-                $defaults[$name] = getenv($envNameMatomo);
-            } elseif (getenv($envName)) {
-                $defaults[$name] = getenv($envName);
+            $envValue = $this->getEnvironmentSetting($name);
+
+            if (!empty($envValue)) {
+                $defaults[$name] = $envValue;
             }
         }
 
+        if (!empty($defaults['password'])) {
+            $defaults['password'] = self::MASKED_PASSWORD_VALUE; // ensure not to show password in UI
+        }
+
         // default values
         $this->addDataSource(new HTML_QuickForm2_DataSource_Array($defaults));
     }
 
+    private function getEnvironmentSetting(string $name)
+    {
+        $envName = 'DATABASE_' . strtoupper($name); // fyi getenv is case insensitive
+        $envNameMatomo = 'MATOMO_' . $envName;
+        if (getenv($envNameMatomo)) {
+            return getenv($envNameMatomo);
+        } elseif (getenv($envName)) {
+            return getenv($envName);
+        }
```

Comment:
```suggestion
        if (getenv($envNameMatomo)) {
            return (string) getenv($envNameMatomo);
        } elseif (getenv($envName)) {
            return (string) getenv($envName);
        }
```

Reply:
`getenv` can in theory return `false|string|array`. I don't think that the rest of the code would handle arrays correctly, but I didn't want to spend any time testing anything around that, so I simply left the code as it was 🙈